.cursor/rules/api-specific-config.mdc

```diff
@@ -57,6 +72,32 @@ alwaysApply: false
   - Use `| null` not `| undefined` for nullable database fields
   - Consider using Result types for operations that may fail
 
+## Atomic Operations & Race Condition Prevention
+
+### Service Layer Atomicity
+- **TOCTOU Prevention**: Avoid Time-Of-Check-Time-Of-Use patterns
```

Comment:
hmm... I'm not sure I understand this one.  At least in these examples, I'm not sure the second one is actually any better than the first, and seems likely to lead to a proliferation of small specialized functions and reduce code re-use